demos/appyx-navigation/iosApp/iosApp/iOSApp.swift

```diff
@@ -1,17 +1,52 @@
 import SwiftUI
 import ios
+import Foundation
+import UIKit
 
 @main
 struct iOSApp: App {
+
+    @UIApplicationDelegateAdaptor(AppDelegate.self)
+    var appDelegate: AppDelegate
+
+    @Environment(\.scenePhase) var scenePhase
+
     var body: some Scene {
         WindowGroup {
             ZStack {
                 Color.white.ignoresSafeArea(.all) // status bar color
-                ContentView()
+                ContentView(lifecycleHelper: appDelegate.lifecycleHolder.lifecycleHelper)
             }
             .onOpenURL { incomingURL in
                 Main_iosKt.handleDeepLinks(url: incomingURL)
             }
+           .onChange(of: scenePhase) { newPhase in
+               switch newPhase {
+               case .background: appDelegate.lifecycleHolder.lifecycleHelper.created()
+               case .inactive: appDelegate.lifecycleHolder.lifecycleHelper.created()
+               case .active: appDelegate.lifecycleHolder.lifecycleHelper.resumed()
+               @unknown default: break
+               }
```

Comment:
Could we move this from client code into some library code / do these in a way that copy-pasting it in every app is not needed?

Reply:
As discussed, we'll try to tackle this in a different PR :)